### PR TITLE
Thin Java Client page added

### PR DESCRIPTION
**Pull request by sergey-serdyukov**
For release 5.5

### CODE REVIEW

**Review by rebekah-lawrence — COMMENTED** [review]
Please sign the CLA.

I have reviewed as far as I can. Please try to apply the spirit of the comments to the remainder of the page.

**Review by oliverhowell — CHANGES_REQUESTED** [review]
Hey Sergey, did you get a chance to review Rebekah's comments and make changes?

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ package-lock.json
 .project
 .settings/
 .vscode/settings.json
+*.html
```
[the remark this answers is not in the document]
Comment (sergey-serdyukov) [reply .gitignore:12]: You are right! This should be removed, it is left over from experiments. Thanks for noticing!

File: docs/modules/clients/pages/java-thin-client.adoc [hunk 2]
```diff
@@ -0,0 +1,1082 @@
...
+
+See the <<java-client-connection-strategy>> section to learn how to configure
+these.
+
+[[configuring-java-client]]
+== Configuring Java Client
+
+=== Declarative vs. Programmatic Configuration
+
+You can configure Hazelcast Thin Java Client declaratively (XML), programmatically (API), or
```
[the remark this answers is not in the document]
Comment (sergey-serdyukov) [reply docs/modules/clients/pages/java-thin-client.adoc:277]: I'll double check. Good point

[the remark this answers is not in the document]
Comment (sergey-serdyukov) [reply docs/modules/clients/pages/java-thin-client.adoc:277]: Or just leave the NOTE with "not supported"?

File: docs/modules/clients/pages/java-thin-client.adoc [hunk 3]
```diff
@@ -0,0 +1,1082 @@
...
+
+[NOTE]
+====
+In the subsequent sections, all examples will be provided using the programmatic approach.
+====
+
+[[client-network]]
+=== Client Network
+
+All network related configuration of Hazelcast Thin Java Client is performed in the class
```
[the remark this answers is not in the document]
Comment (sergey-serdyukov) [reply docs/modules/clients/pages/java-thin-client.adoc:304]: Java Thin Client sounds good unless someone comes up with a better name

File: docs/modules/clients/partials/nav.adoc [hunk 4]
```diff
@@ -1,6 +1,7 @@
 * Connecting to a Cluster
 ** xref:clients:hazelcast-clients.adoc[]
 ** xref:clients:java.adoc[]
+** xref:clients:java-thin-client.adoc[]
```
[the remark this answers is not in the document]
Comment (sergey-serdyukov) [reply docs/modules/clients/partials/nav.adoc]: Done

File: docs/modules/clients/pages/java-thin-client.adoc [hunk 5]
```diff
@@ -0,0 +1,1082 @@
+= Thin Java Client
```
[the remark this answers is not in the document]
Comment (sergey-serdyukov) [reply docs/modules/clients/pages/java-thin-client.adoc]: Planning to split the page in the nearest future.

File: docs/modules/clients/pages/java-thin-client.adoc [hunk 6]
```diff
@@ -0,0 +1,1082 @@
...
+Instead of giving up, throwing an exception and stopping the client,
+the client retries to connect as configured which is described in the
+<<configuring-client-connection-retry, Configuring Client Connection Retry section>>.
+
+The client executes each operation through the already established connection to the cluster.
+If this connection(s) disconnects or drops, the client tries to reconnect as configured.
+
+The initial connection of a client is established using one of the addresses you provide in the <<configuring-address-list, address list>>. 
+The client gets the addresses of other members in the cluster from the initially connected member.
+If your client is <<java-client-operation-modes, smart>>, the client tries to connect to them.
```
[the remark this answers is not in the document]
Comment (sergey-serdyukov) [reply docs/modules/clients/pages/java-thin-client.adoc]: It has been removed. In the future we will review the usage of connection modes.

**Comment by sergey-serdyukov** [comment]
> Hey Sergey, did you get a chance to review Rebekah's comments and make changes?

Hi @oliverhowell , yes Rebekah's comments were helpful, all changes have been made. I'll close this PR to simplify rebase as it was opened long ago and there are many other changes. Gonna to send another PR shortly (with the same changes).

**Pull request closed without merging**